backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/MicrometerPerformanceMonitoringService.java

```diff
@@ -42,58 +44,92 @@
 public class MicrometerPerformanceMonitoringService implements PerformanceMonitoringService
 {
 	private final MeterRegistry meterRegistry;
-	private final Optional<APMPerformanceMonitoringService> apmPerformanceMonitoringService;
+	private final ThreadLocal<Integer> depth = ThreadLocal.withInitial(() -> 0);
+	private final ThreadLocal<String> initiator = ThreadLocal.withInitial(() -> "");
+	private final ThreadLocal<String> initiatorWindow = ThreadLocal.withInitial(() -> "");
+	private final ThreadLocal<ArrayList<String>> calledBy = ThreadLocal.withInitial(() -> new ArrayList<>());
+	private final ThreadLocal<Boolean> isInitiatorLabelActive = ThreadLocal.withInitial(() -> false);
+
+	@Value("${performance.monitoring.enable:false}")
+	private Boolean perfMonEnvVar;
 
 	public MicrometerPerformanceMonitoringService(
-			@NonNull final Optional<APMPerformanceMonitoringService> apmPerformanceMonitoringService,
 			@NonNull final MeterRegistry meterRegistry)
 	{
-		this.apmPerformanceMonitoringService = apmPerformanceMonitoringService;
 		this.meterRegistry = meterRegistry;
 	}
 
 	@Override
-	public <V> V monitorTransaction(
+	public <V> V monitor(
 			@NonNull final Callable<V> callable,
-			@NonNull final TransactionMetadata metadata)
+			@NonNull final PerformanceMonitoringService.Metadata metadata)
 	{
+		if(perfMonEnvVar = false)
```

Comment:
> perfMonEnvVar = false

that will be always false :)
you wanted to use "==" but ... better, pls use sysconfig